main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -1242,6 +1309,75 @@ object Lowering {
     mkTag(Enums.HeadTerm, s"App$arity", innerExp, Types.HeadTerm, loc)
   }
 
+  /**
+    * Make the array of MpmcAdmin objects which will be passed to `selectFrom`
+    */
+  private def mkChannelAdminArray(rs: List[SelectChannelRule], channels: Array[(Symbol.VarSym, Expression)], pur: Type, eff: Type, loc: SourceLocation): Expression = {
+    val admins = rs.zipWithIndex map {
+      case (SelectChannelRule(_, c, _), i) =>
+        val admin = Expression.Def(Defs.ChannelMpmcAdmin, Type.mkPureArrow(c.tpe, Types.ChannelMpmcAdmin, loc), loc)
+        Expression.Apply(admin, List(Expression.Var(channels(i)._1, c.tpe, loc)), Types.ChannelMpmcAdmin, pur, eff, loc)
```

Comment:
Instead of `channels(i)._1`, could we zip together `rs`, `channels`, and `zipWithIndex`?